app/src/main/java/net/squanchy/schedule/filterschedule/FilterSchedule.kt

```diff
@@ -0,0 +1,135 @@
+package net.squanchy.schedule.filterschedule
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.os.Bundle
+import android.support.annotation.ColorInt
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.CheckBox
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.functions.BiFunction
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_track_filters.*
+import net.squanchy.R
+import net.squanchy.schedule.domain.view.Track
+import net.squanchy.service.repository.TracksRepository
+
+class FilterScheduleActivity : AppCompatActivity() {
+
+    private lateinit var tracksRepository: TracksRepository
+    private lateinit var tracksFilter: TracksFilter
+    private lateinit var trackAdapter: TrackFiltersAdapter
+
+    private var subscription: Disposable? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_track_filters)
+
+        val component = tracksFilterComponent(this)
+        tracksRepository = component.tracksRepository()
+        tracksFilter = component.tracksFilter()
+
+        trackAdapter = TrackFiltersAdapter(this)
+
+        trackFiltersList.layoutManager = LinearLayoutManager(this)
+        trackFiltersList.adapter = trackAdapter
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        subscription = Observable.combineLatest(tracksRepository.tracks(), tracksFilter.selectedTracks, combineIntoCheckableTracks())
+            .subscribeOn(Schedulers.computation())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                trackAdapter.updateTracks(it) { track, selected ->
+                    val selectedTracks = it.filter { it.second }.map { it.first }.toSet()
```

Comment:
I'm a bit confused by all the `it`s, can we rename/deconstruct and rename?

Reply:
94d39ffe3f0467a901c81cf79b5d1e8eafb92e40